flink-runtime/src/test/java/org/apache/flink/runtime/jobmanager/scheduler/SchedulerIsolatedTasksTest.java

```diff
@@ -212,14 +154,12 @@ public void testScheduleWithDyingInstances() throws Exception {
 
 		// cannot get another slot, since all instances are dead
 		try {
-			testingSlotProvider.allocateSlot(new ScheduledUnit(getDummyTask()), false, SlotProfile.noRequirements(), TestingUtils.infiniteTime()).get();
+			testingSlotProvider
+				.allocateSlot(new ScheduledUnit(getDummyTask()), SlotProfile.noRequirements(), TestingUtils.infiniteTime())
+				.get(2000L, TimeUnit.MILLISECONDS);
```

Comment:
This change looks a bit weird. I think we only see the `TimeoutExeception` because of the `get(2000L, TimeUnit.MILLISECONDS)` and not from the `testingSlotProvider` because we use the `ComponentMainThreadExecutorServiceAdapter.forMainThread` which swallows the timeout of the slot allocation. I suggest to change this and to set a very low slot allocation timeout to fail fast.

Reply:
I verify with the diff below

```diff
@@ -155,11 +155,11 @@ public class SchedulerIsolatedTasksTest extends SchedulerTestBase {
                // cannot get another slot, since all instances are dead
                try {
                        testingSlotProvider
-                               .allocateSlot(new ScheduledUnit(getDummyTask()), SlotProfile.noRequirements(), TestingUtils.infiniteTime())
-                               .get(2000L, TimeUnit.MILLISECONDS);
+                               .allocateSlot(new ScheduledUnit(getDummyTask()), SlotProfile.noRequirements(), Time.milliseconds(0L))
+                               .get();
                        fail("Scheduler served a slot from a dead instance");
-               } catch (TimeoutException e) {
-                       // expected
+               } catch (Exception e) {
+                       assertTrue(e.getCause() instanceof TimeoutException);
                }
 
                // now the latest, the scheduler should have noticed (through the lazy mechanisms)
```

However, due to we `get` in main thread and schedule timeout future in main thread, we dead lock by main thread itself...Maybe we introduce a new test with `ComponentMainThreadExecutorServiceAdapter.forSingleThreadExecutor`. I will appreciate it if you can help port this test.

Reply:
Yes, I think the test should not use the `ComponentMainThreadExecutorServiceAdapter.forMainThread`. Let me try to port the test for you.